sink/datadog.go

```diff
@@ -367,8 +367,13 @@ func (s *Datadog) sendApi(ddCtx context.Context, dp []datadogV2.MetricSeries) er
 			rangeEnd = len(dp)
 		}
 
+		optParams := *datadogV2.NewSubmitMetricsOptionalParameters()
+		if s.compress {
+			optParams.ContentEncoding = datadogV2.METRICCONTENTENCODING_GZIP.Ptr()
+		}
+
 		if _, r, err := s.metricsApi.SubmitMetrics(ddCtx, *datadogV2.NewMetricPayload(dp[rangeStart:rangeEnd]), *datadogV2.NewSubmitMetricsOptionalParameters()); err != nil {
-			if r.StatusCode == http.StatusRequestEntityTooLarge {
+			if r != nil && r.StatusCode == http.StatusRequestEntityTooLarge {
```

Comment:
Yeah when `err` is nil, there's usually no guarantee on the response. This is how built-in `http` works, at least.